.changeset/khaki-apes-repeat.md

```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/ui': patch
+---
+
+fix infinite loop on refresh after refresh token expires
```

Comment:
I think this patch description could use more information. We may want to reference it later in the changelog, and it's pretty vague

Reply:
+1!

```suggestion
This patch ensures refresh tokens are handled properly after expiration. 

Refresh tokens are used to refresh your idToken and accessToken. While it's uncommon for refresh tokens to expire during a lifecycle of app due to their longevity (default 30 days), this would cause an early `tokenRefresh_failure` event on refresh which previously confused the authenticator. This is now handled properly from the Authenticator.
```

Reply:
Maybe link the issue number here can help?

Reply:
@zchenwei, @reesscot

Updated -- see my latest revision.